README.md

```diff
@@ -1 +1,167 @@
-# node-non-blocking-callback
\ No newline at end of file
+# Callback Hell
+
+## Overview
+
+We've used callbacks in previous units to create an easier workflow and schedule events to happen in the future, as opposed to our code being executed in a linear order. While callbacks are incredibly useful, they can get pretty complicated pretty quickly.
+
+Let's say you have to implement three HTTP requests to a third-party service (like Facebook or Google), but they need to be performed in a sequential order, because the second and third request use information from the response of the first request (e.g., API token).  We can't delay code the way we do in traditional synchronous style, so you end up with nested callbacks to ensure the right order of execution. This is the pseudo code:
+
+```js
+callOne({...}, function(error, data1) {
+    callTwo(data1, function(error, data2) {
+        callThree(data2, function(error, data3) {
+        ...
+        })
+    })
+})
+```
+
+Remember, this is asynchronous world so we cannot type `callOne` then `callTwo` on a new line. We need to use callback and with callbacks (the way they are written in the example above) comes indentation.
+
+Now add on top of that a few callbacks to perform something else essential like opening a database connection and starting a server. Look at this:
+
+
+```js
+fs.readdir(source, function (err, files) {
+  if (err) {
+    console.log('Error finding files: ' + err)
+  } else {
+    files.forEach(function (filename, fileIndex) {
+      console.log(filename)
+      gm(source + filename).size(function (err, values) {
+        if (err) {
+          console.log('Error identifying file size: ' + err)
+        } else {
+          console.log(filename + ' : ' + values)
+          aspect = (values.width / values.height)
+          widths.forEach(function (width, widthIndex) {
+            height = Math.round(width / aspect)
+            console.log('resizing ' + filename + 'to ' + height + 'x' + height)
+            this.resize(width, height).write(dest + 'w' + width + '_' + filename, function(err) {
+              if (err) console.log('Error writing file: ' + err)
+            })
+          }.bind(this))
+        }
+      })
+    })
+  }
+}
+```
+
+You might end up with something called the nested approach, pyramid of doom or callback hell. This code is very hard to read and maintain. People who have to work on this code after you might start to hate you!
```

Comment:
This comes across as a bit harsh. Maybe instead of talking about coworkers throwing shade, we can focus on the difficulties introduced by callback hell -- and the fact that even the person who wrote bad code, coming back to it a week or two later, will be dismayed.